.gitlab-ci.yml

```diff
@@ -790,3 +817,77 @@ cleanup_eks_cluster:success:
 cleanup_eks_cluster:failed:manual:
   when: manual
   <<: *eks_cleanup
+
+changelog:
+  image: ${CI_DEPENDENCY_PROXY_DIRECT_GROUP_IMAGE_PREFIX}/node:20
+  stage: changelog
+  tags:
+    - hetzner-amd-beefy
+  rules:
+    - if: $CI_PIPELINE_SOURCE == "pipeline"
+      when: never
+    - if: $RUN_RELEASE == "true"
+      when: never
+    - if: $CI_COMMIT_BRANCH =~ "/^\d+\.\d+\.x$/"
+    - if: $CI_COMMIT_BRANCH == $DEFAULT_BRANCH
+  before_script:
+    # install release-please
+    - npm install -g release-please
+    # install github-cli
+    - |
+      mkdir -p -m 755 /etc/apt/keyrings \
+        && wget -qO- https://cli.github.com/packages/githubcli-archive-keyring.gpg | tee /etc/apt/keyrings/githubcli-archive-keyring.gpg > /dev/null \
+        && chmod go+r /etc/apt/keyrings/githubcli-archive-keyring.gpg \
+        && echo "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" | tee /etc/apt/sources.list.d/github-cli.list > /dev/null \
+        && apt update \
+        && apt install gh jq -y
+    # Setting up git
+    - |
+      echo "INFO - setting up git"
+      git config --global user.email "${GITHUB_USER_EMAIL}"
+      git config --global user.name "${GITHUB_USER_NAME}"
+    - npm install -g git-cliff
+    # GITHUB_TOKEN for Github cli authentication
+    - export GITHUB_TOKEN=${GITHUB_CLI_TOKEN}
+    # getting the centralized git cliff config
+    - wget --output-document cliff.toml https://raw.githubusercontent.com/mendersoftware/mendertesting/master/utils/cliff.toml
+  script:
+    - release-please release-pr
+        --token=${GITHUB_BOT_TOKEN_REPO_FULL}
+        --repo-url=${GITHUB_REPO_URL}
+        --target-branch=${CI_COMMIT_REF_NAME}
+    # git cliff: override the changelog
+    - test $GIT_CLIFF == "false" && echo "INFO - Skipping git-cliff" && exit 0
+    - git clone https://${GITHUB_USER}:${GITHUB_BOT_TOKEN_REPO_FULL}@github.com/${GITHUB_REPO_URL}
+    - cd ${GITHUB_REPO_URL#*/}
+    - RELEASE_PLEASE_PR=$(gh pr list --author "${GITHUB_USER_NAME}" --head "release-please--branches--${CI_COMMIT_REF_NAME}" --json number | jq -r '.[0].number // empty')
+    - test -z "$RELEASE_PLEASE_PR" && echo "No release-please PR found" && exit 0
+    - echo "INFO - generating changelog notes from git cliff"
+        && gh pr checkout $RELEASE_PLEASE_PR
+        && git cliff --output mender/CHANGELOG.md --bump --github-repo ${GITHUB_REPO_URL}
+        && git add mender/CHANGELOG.md
+        && git commit --amend -s --no-edit
+        && git push origin --force
+    # note: using sed to prevent release-please to mess up with the Github Release body
+    - echo "INFO - updating PR body"
+        && git cliff --unreleased --bump -o tmp_pr_body.md --github-repo ${GITHUB_REPO_URL}
+        && sed -i 's/## mender-\([0-9.]*\)/## \1/' tmp_pr_body.md
+        && gh pr edit $RELEASE_PLEASE_PR --body-file tmp_pr_body.md
+        && rm tmp_pr_body.md
```

Comment:
Every time I read this script I keep wondering what's the purpose of release-please. Isn't it simply a glorified `gh pr create`/`gh release create`?
If I'm not mistaken, this snippet is forcefully overwriting the entire content of the PR.

Reply:
Added the `--unreleased` flag to just add the latest release, without overriding the previous ones.

> Every time I read this script I keep wondering what's the purpose of release-please. Isn't it simply a glorified gh pr create/gh release create?

That, but most importantly it has the logic to check every commit and understand which will be the next version; the logic also applies to maintenance branches. 
We could just rewrite this logic implementing a git cliff wrapper in bash/python, but this will end up in a custom tool, something that we'd like to avoid. The tool could be pretty simple, but would still require a fair amount of code and cases.

OTOH, we're still considering release-please experimental, so if the experience is these "testing" repositories is bad, we could simply try with another (internal?) tool.